controller/src/test/java/io/pravega/controller/store/stream/StreamTest.java

```diff
@@ -0,0 +1,129 @@
+        testStream(stream);
+    }
+
+    private void testStream(PersistentStreamBase<Integer> stream) throws InterruptedException, ExecutionException {
+        long creationTime1 = System.currentTimeMillis();
+        long creationTime2 = creationTime1 + 1;
+        final ScalingPolicy policy1 = ScalingPolicy.fixed(5);
+        final ScalingPolicy policy2 = ScalingPolicy.fixed(6);
+
+        final StreamConfiguration streamConfig1 = StreamConfiguration.builder().scope("test").streamName("test").scalingPolicy(policy1).build();
+        final StreamConfiguration streamConfig2 = StreamConfiguration.builder().scope("test").streamName("test").scalingPolicy(policy2).build();
+
+        CreateStreamResponse response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        stream.storeCreationTimeIfNotExists(creationTime1).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime2).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+
+        stream.createConfigurationIfNotExists(streamConfig1).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime2).get();
+        assertEquals(CreateStreamResponse.CreateStatus.EXISTS_CREATING, response.getStatus());
+
+        stream.createStateIfNotExists(State.UNKNOWN).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime2).get();
+        assertEquals(CreateStreamResponse.CreateStatus.EXISTS_CREATING, response.getStatus());
+
+        stream.updateState(State.CREATING).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
```

Comment:
We get NEW because we are using the same "creationTime".
Since the caller has given the same creation time [Note: creation time is generated in the controller and not supplied by end user], so we treat this as new Stream.] 
Note 2: we will still use the configuration that was stored earlier. 

Imagine a task running this step idempotently supplying the same "create time", we do not want to return StreamExists.
This creation time check ensures that is not the case.